stone/backend.py

```diff
@@ -40,28 +41,14 @@ def remove_aliases_from_api(api):
         # Remove nested aliases first. This way, when we replace an alias with
         # its source later on, it too is alias free.
         for alias in namespace.aliases:
-            data_type = alias
-            while True:
-                # For better or for worse, all non-user-defined types that
-                # reference other types do so with a 'data_type' attribute.
-                if hasattr(data_type, 'data_type'):
-                    if is_alias(data_type.data_type):
-                        # Skip the alias (this looks so terrible...)
-                        data_type.data_type = data_type.data_type.data_type
-                    data_type = data_type.data_type
-                else:
-                    break
-
+            # For better or for worse, all non-user-defined types that
+            # reference other types do so with a 'data_type' attribute.
+            if hasattr(alias, 'data_type'):
+                resolve_aliases(alias.data_type)
         for data_type in namespace.data_types:
             for field in data_type.fields:
-                data_type = field
-                while True:
-                    if hasattr(data_type, 'data_type'):
-                        if is_alias(data_type.data_type):
-                            data_type.data_type = data_type.data_type.data_type
-                        data_type = data_type.data_type
-                    else:
-                        break
+                if hasattr(field, 'data_type'):
+                    field.data_type = resolve_aliases(field.data_type)
```

Comment:
Since we already have all aliases resolved in above code, maybe it's easier to just write (knowing there is no intermediate alias):
```
if is_alias(field.data_type):
    field.data_type = field.data_type.data_type
```

This aligns with how route args are handled below.